tensorflow/compiler/tf2xla/kernels/reduction_ops_common.cc

```diff
@@ -122,7 +129,7 @@ void XlaReductionOp::Compile(XlaOpKernelContext* ctx) {
   xla::XlaComputation reduction_computation = r.Build().ConsumeValueOrDie();
 
   auto reduce = xla::Reduce(data, initial, reduction_computation, xla_axes);
-  auto finalized = BuildFinalizer(b, data, reduce, xla_axes);
+  auto finalized = BuildFinalizer(b, ctx->Input(0), reduce, xla_axes);
```

Comment:
Why did this need to change?  Earlier we were passing in the result of `ConvertElementType` right?

Reply:
You're right. My original thought is to make codes unchanged in this part, so maybe we should do something like
```cpp
  auto input = xla::ConvertElementType(ctx->Input(0), type);
  auto data = PreprocessInput(b, input);
  ...
  auto finalized = BuildFinalizer(b, input, reduce, xla_axes);
```
What do you think about this? Thanks!

Reply:
LGTM, except that I'd prefer s/input/converted_input/